docs/user/faq.rst

```diff
@@ -617,19 +617,17 @@ now simply set attributes on the object:
    # Falcon 2.0
    req.context.cache_backend = MyUltraFastCache.connect()
 
-Since the default context type derives from dict, the existing code will work
-unmodified with Falcon 2.0. Nevertheless, it is recommended to change usage
-as outlined above since the ability to use context as dictionary may be
-removed in a future release.
+The new default context type emulates the dict interface in a way that context
+attributes are linked to dict items, i.e. setting an object attribute also sets
+the corresponding dict item, and vice versa. As a result, the existing code
+will largely work unmodified with Falcon 2.0. Nevertheless, it is recommended
+to change usage as outlined above since the ability to use context as dict may
```

Comment:
I would suggest emphasizing the interface (mapping) vs. the type (dict) to avoid any confusion. With that in mind, you might change this:

> since the ability to use context as dict may be removed in a future release.

To this:

> since the dict-like mapping interface may be removed from the context type in a future release.